core/src/main/scala/kafka/log/Log.scala

```diff
@@ -133,6 +135,24 @@ class Log(val dir: File,
     },
     tags)
 
+  newGauge("LogSegments",
+    new Gauge[util.List[String]] {
+      def value = {
+        val list = logSegments.toSeq.map { seg =>
+          s"baseOffset=${seg.baseOffset}, created=${seg.created}, logSize=${seg.size}, indexSize=${seg.index.sizeInBytes()}"
+        }
+        // Explicitly returning Java list to support JMX clients that don't have Scala runtime in the classpath
+        new util.ArrayList[String](list.asJava)
```

Comment:
The jmx metric value will use comma to delimit different segments. Since each segment also has comma separated fields, it makes it hard to parse the text value. Could we use a different delimiter within a segment?